### DISCUSSION TITLE
cpp2decl: make cpp2 greppable

### DISCUSSION THREAD

**Discussion by nigeltao** [hsutter/cppfront#657] [Show and tell]
This does for cpp2 declarations what [gron](https://github.com/tomnomnom/gron) does for JSON.

See https://github.com/nigeltao/cppfront/commit/d81a36c8195ee69c9d29d6221d8824b6805fc7d2

---

When passing https://github.com/hsutter/cppfront/blob/1764f0ccbefc24e93b583cc67a73aafd9513a299/regression-tests/pure2-types-inheritance.cpp2 as input:

```
// cpp2decl prints the declarations in a cpp2 file. For example, running on
// regression-tests/pure2-types-inheritance.cpp2 (at commit 1764f0c) prints:
//
// Human                           : type
// Human::speak                    : (this)
// Human::operator=                : (move this)
// N::Machine                      : type
// N::Machine::operator=           : (out this, _: std::string)
// N::Machine::work                : (this)
// N::Machine::operator=           : (move this)
// Cyborg                          : type
// Cyborg::name                    : std::string
// Cyborg::this                    : Human
// Cyborg::address                 : std::string
// Cyborg::this                    : N::Machine<(expression)>
// Cyborg::operator=               : (out this, n: std::string)
// Cyborg::speak                   : (this)
// Cyborg::work                    : (this)
// Cyborg::print                   : (this)
// Cyborg::operator=               : (move this)
// make_speak                      : (h: Human)
// do_work                         : (m: N::Machine<(expression)>)
// main                            : ()
```

**Comment by JohelEGP** [comment]
I can't help but notice that `cpp2decl` doesn't mean "cpp" to "decl",
as it usually does for conversion tools (e.g. `pdf2txt`).
Perhaps that's just an unfortunate side-effect of being "2".
Maybe someone with previous experience working on a "2" knows how to work around this issue,
or if there's actually any issue at all.

**Reply by gregmarr** [reply]
`cpp2_decls`, `cpp2_get_decls`, `cpp2getdecls`?

**Reply by nigeltao** [reply]
We could rename cpp2 to cxxi: a 21st century language in the C/C++ lineage. XXI is 21 in Roman numerals. C++1 files sometimes end with .cxx.

**Reply by hsutter** [reply]
I appreciate the alternative name suggestions. Funnily enough, just yesterday I had someone else independently email me a suggestion based on Roman numerals: "CPPii" !

Since I already wrote a longer reply to that person with other rationale, I figured this was a good time write a Design Note about the Cpp2 name: [Design note: Cpp2 name](https://github.com/hsutter/cppfront/wiki/Design-note%3A-Cpp2-name)
